server/bleep/src/webserver/answer.rs

```diff
@@ -1092,3 +1085,52 @@ impl StopWatch {
         duration
     }
 }
+
+fn dot(a: &[f32], b: &[f32]) -> f32 {
```

Comment:
Can you move these functions into `semantic.rs`. They are not specific to the `answer` pipeline and could be applied to any set of semantic search results.